### ISSUE TITLE
Sensors w/o UoM: Logbook availability issues

### ISSUE THREAD

**Issue by ildar170975** [home-assistant/frontend#18847]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Consider this template sensor:
```
template:
  - sensor:
      - name: ...
        state: >-
          {{ states('input_select.test_value') }}
```
which may be switched between values like `one`, `two`, `three` etc.
It is possible to see a Logbook for this sensor:

![изображение](https://github.com/home-assistant/frontend/assets/71872483/ad563815-d17e-41ad-a32d-00907f138165)

But the Logbook is not available in more-info:

![изображение](https://github.com/home-assistant/frontend/assets/71872483/8382e661-0d52-4a9c-8b7b-c19921a4de85)

which seems to be not correct.

---------

Another case.
Consider a sensor with `device_class: aqi` or `ph`.
These classes do not have UoM - but values are numerical.
But these sensors may be selected in Logbook:

![изображение](https://github.com/home-assistant/frontend/assets/71872483/3fee8fe3-a851-4f99-bd4b-ae9065c585ee)

It seems to be inconsistent to have a Logbook available & chartline graph at same time:

![изображение](https://github.com/home-assistant/frontend/assets/71872483/c5035912-798f-48f4-9c39-e3c3e652f3f6)

Although the `aqi` & `ph` sensors' more-info do not have a logbook displayed.

Proposals:
1. If an entity may be selected on a Logbook page (or for a Logbook card) - always show a Logbook in more-info as well.
2. A `sensor` entity could be selected on a Logbook page (or for a Logbook card) if this entity:
- is not numerical;
- does not have UoM;
- either does not have `device_class` or has a `device_class: enum`.

### Describe the behavior you expected

as above

### Steps to reproduce the issue

as above

### What version of Home Assistant Core has the issue?

2023.11.2

### In which browser are you experiencing the issue with?

Chrome 119.0.6045.160 

### Which operating system are you using to run this browser?

Win10x64

**Comment by sjd17** [comment]
I have the same request

**Comment by ildar170975** [comment]
up

**Comment by melyux** [comment]
Me too

**Comment by JCalvi** [comment]
I would like this also.

**Comment by sjd17** [comment]
Merry Christmas

**Comment by melyux** [comment]
This looks to be fixed now on 2025.2, everyone check it out.

**Comment by sjd17** [comment]
It is! Awesome! Already proved useful too. I checked by looking at my wifi network logbook and saw a bunch of short wifi connections while driving that were never visible in history.

**Comment by melyux** [comment]
But but but @Petro31 [assured us](https://community.home-assistant.io/t/wth-why-dont-sensors-show-logbook-in-dialog/476831/40) this bug was actually a feature, like he does with most bugs on his forums.

**Comment by ildar170975** [comment]
It may be fixed in SOME extend.
But here is what I see with some input_select value:

1. Select some `input_select` from from Settings->Entities & start changing it's value.
A logbook in more-info is filled with some records - CORRECT:
![Image](https://github.com/user-attachments/assets/0ea243f6-9416-4c6b-b2f7-175ddaff095d)

2. Open the Logbook page in a different browser's tab & select this entity.
The Logbook page does not have records - WRONG.
Note that this entity is not excluded from Recorder.
![Image](https://github.com/user-attachments/assets/4e29d176-df83-4e6c-bafd-1d22b6bbb810)

3. Add a new Logbook card for this entity.
No records - WRONG.
Save the card & close the UI editor.
![Image](https://github.com/user-attachments/assets/48a92ef3-9347-45ed-a282-1d781866f781)

4. The added card still does not have records - WRONG:
![Image](https://github.com/user-attachments/assets/55fd299b-8530-433f-ac9c-588acb8a8034)

5. Press F5 just in case.
The card gets an loading indicator (could be infinite).
A "top" command in cli (debian) shows a very low CPU usage.
![Image](https://github.com/user-attachments/assets/dd5866c3-e4c1-43ec-aa64-a59d516c417b)

6. Open more-info again, change a value.
Then go to the Logbook page - now it has some records, but they are not all of them, only recent are displayed (WRONG):
![Image](https://github.com/user-attachments/assets/bb3ecf89-6973-410c-96c1-df2c0f75bfa4)

7. Go to the logbook card - it has some records displayed (again, not all of them, only recent ones):
![Image](https://github.com/user-attachments/assets/4d94c5af-fe55-4090-accd-fad299e7453a)

8. Open UI editor - again it is empty (WRONG):
![Image](https://github.com/user-attachments/assets/9a1eeaf3-d954-42eb-b6e5-f1e02da2cf42)

9. Close the UI editor, press F5 - you may either get an loading indicator or an empty logbook or some records (not all of them). I would call it ERRATIC.

Here is a gif, if you patient enough you will see that sometime the card has some messages but in most cases it is empty:
![Image](https://github.com/user-attachments/assets/42800ae3-5779-4a19-bfd1-c6257b4dbe32)

**Comment by Petro31** [comment]
Logbook entries won't show up for numeric sensors.  That's by design IIRC

**Comment by raldred** [comment]
I wonder why? The graph only shows the value, it doesn't show who or what changed the value, which is also useful. Particularly when automations are involved.

**Comment by raldred** [comment]
Indeed, it's a number entity, inherited from `NumberEntity`
defined in `number.py`

**Comment by Petro31** [comment]
> Really?
> The person meant a number entity, not a sensor.

The title of this issue is `Sensors w/o UoM`... If they are referring to a different entity type, a new issue should be created.

**Comment by ildar170975** [comment]
> The title of this issue is Sensors w/o UoM... If they are referring to a different entity type, a new issue should be created. All domains/platforms are not the same.

Agree, but your post was right after a user's post describing a problem "input number: logbook sometimes is shown, sometimes not" - that is why I thought that you reacted at that "input number" problem.

**Comment by raldred** [comment]
@Petro31 Do you have any reference to the reason for why number entities with UoM are being excluded from tracking logbook entries?
I'm happy to open an issue but if there's an agreed reasoning, then it would be helpful to understand that before trying to challenge the status quo.

**Comment by Petro31** [comment]
There won't be documentation for something that obscure.

**Comment by raldred** [comment]
Ok, I'm surprised, is the logbook not an important core feature? Number entities lacking support for it seems odd. It seems it's asked often on the forum as to why, usually if something is repeatedly asked, further clarification of behaviour would be helpful so that user's aren't continuously caught out, don't you think?

**Comment by raldred** [comment]
Fair enough, appreciate your insight.

**Comment by karwosts** [comment]
Looking more at this it's definitely a bit of a mess, across both core and frontend. I catalogued the various behavior across all the relevant components. 

| Domain  | UoM | Numeric DC[^2] | StateClass | Core [^3] | Panel[^4] | More-Info[^5] | Comment |
|---------|-----|------------|------------|----------|-------------|-----------|---------|
| Sensor  | Y   | -          | -          | N        | N           | N         |   ✔️       | 
| Sensor  | N   | Y          | -          | Y        | Y           | N         |   ❌ Frontend partially uses the numeric device classes to suppress the logbook.  Should probably be disabled everywhere like state_class.  |
| Sensor  | N   | -          | Y          | N        | N           | N         |    ✔️     |
| Sensor  | N   | N          | N          | Y        | Y           | Y         |  ✔️        |
| Number[^1]  | Y   | -          | -          | N        | Y           | Y         |  ❌Number should probably have the same behavior whether it has a UoM or not, but not exactly sure which way it should go. I can see disabling it as it is fully numeric, though on the same time it is a user-activated control, which may be more interesting to see changes in logbook form.   |
| Number  | N   | -          | -          | Y        | Y           | Y         |  ❌ See above.      |
| Counter | -   | -          | -          | N        | Y           | N         |  ❌ Seems like should be N?       |
| Zone    | -   | -          | -          | Y        | Y           | Y         |  ❌ Seems like should be N?      |


[^1]: `number` or `input_number`
[^2]: Numeric device_class like `pH` or `AQI`
[^3]: `Y` if core API will return any values for this combination. `N` if it always returns empty.
[^4]: If the entity is selectable in the logbook panel.
[^5]: If entity will show a logbook section in more-info.


I would be willing to look at trying to harmonize these across all implementations, though some questions remain as to what the expected proper behavior should be.
